### PR TITLE
[as5835-54x] Add to support API2.0

### PR DESCRIPTION
**Pull request by jostar-yang**
**- Why I did it**
Support API2.0
**- How I did it**
Implement API2.0 needed code
**- How to verify it**
Test each API.
root@sonic:/home/admin# ./test_as5835-54x_api2.py
DEBUG:root:Test Chassis
DEBUG:root:get_name=Accton-AS5835-54X
DEBUG:root:get_base_mac=80:A2:35:D2:46:B5
DEBUG:root:get_serial_number=583554X1922008
DEBUG:root:get_system_eeprom_info={'0xFE': '0x54E5C8B0', '0x28': 'x86_64-accton_as5835_54x-r0', '0x29': '2019.08.00.01', '0x22': 'FP1ZZ56540B7A', '0x23': '583554X1922008', '0x21': '5835-54X-O-AC-F', '0x27': 'R0BA', '0x24': '80:A2:35:D2:46:B5', '0x25': '06/21/2019', '0x2B': 'Accton', '0x2C': 'TW', '0x2A': '256', '0x2D': 'Edgecore', '0x2E': 'b.b.1.0'}
DEBUG:root:get_reboot_cause=('REBOOT_CAUSE_NON_HARDWARE', "User issued 'reboot' command [User: admin, Time: Wed 13 Jan 2021 04:50:43 PM UTC]")
DEBUG:root:
Test Component
DEBUG:root:get_num_components=4
DEBUG:root:get_name=CPLD1
DEBUG:root:get_name=CPLD2
DEBUG:root:get_name=CPLD3
DEBUG:root:get_name=BIOS
DEBUG:root:get_firmware_version0 =2
DEBUG:root:get_firmware_version1 =2
DEBUG:root:get_firmware_version2 =2
DEBUG:root:get_firmware_version3 =AS5835 V37 20190117
DEBUG:root:get_description=CPLD 1
DEBUG:root:get_description=CPLD 2
DEBUG:root:get_description=CPLD 3
DEBUG:root:get_description=Basic Input/Output System
DEBUG:root:
Fan test
DEBUG:root:fan-1, presence=1
DEBUG:root:fan 1 direction=exhaust
DEBUG:root:fan 1 speed()=40
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-2, presence=1
DEBUG:root:fan 2 direction=exhaust
DEBUG:root:fan 2 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-3, presence=1
DEBUG:root:fan 3 direction=intake
DEBUG:root:fan 3 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-4, presence=1
DEBUG:root:fan 4 direction=intake
DEBUG:root:fan 4 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-5, presence=1
DEBUG:root:fan 5 direction=intake
DEBUG:root:fan 5 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-6, presence=1
DEBUG:root:fan 6 direction=intake
DEBUG:root:fan 6 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-7, presence=1
DEBUG:root:fan 7 direction=intake
DEBUG:root:fan 7 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=40
DEBUG:root:fan-8, presence=1
DEBUG:root:fan 8 direction=intake
DEBUG:root:fan 8 speed()=40
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-9, presence=1
DEBUG:root:fan 9 direction=intake
DEBUG:root:fan 9 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:fan-10, presence=1
DEBUG:root:fan 10 direction=intake
DEBUG:root:fan 10 speed()=50
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=50
DEBUG:root:
PSU test
DEBUG:root:PSU-1
DEBUG:root:presence=1
DEBUG:root:get_name=PSU-1
DEBUG:root:get_powergood=1
DEBUG:root:get_voltage=11.86
DEBUG:root:get_current=7.17
DEBUG:root:get_power=81.00
DEBUG:root:get_temperature=31.000
DEBUG:root:get_voltage_high_threshold=1472.00
DEBUG:root:get_voltage_low_threshold=-768.00
DEBUG:root:psu:get_num_fans=1
DEBUG:root:psu:get_fan().get_presence=1
DEBUG:root:dir_str=/sys/bus/i2c/devices/11-0058/psu_fan_dir, val=F2B
DEBUG:root:psu:get_fan().get_direction=exhaust
DEBUG:root:psu:get_fan().get_speed=12
DEBUG:root:psu:get_num_thermals()=0
DEBUG:root:psu:get_temperature=31.000
DEBUG:root:PSU-2
DEBUG:root:presence=0
DEBUG:root:get_name=PSU-2
DEBUG:root:get_powergood=0
DEBUG:root:get_voltage=0.00
DEBUG:root:get_current=0.00
DEBUG:root:get_power=0.00
DEBUG:root:get_temperature=0.000
DEBUG:root:get_voltage_high_threshold=0.00
DEBUG:root:get_voltage_low_threshold=0.00
DEBUG:root:psu:get_num_fans=1
DEBUG:root:psu:get_fan().get_presence=1
DEBUG:root:dir_str=/sys/bus/i2c/devices/12-005b/psu_fan_dir, val=
DEBUG:root:psu:get_fan().get_direction=intake
DEBUG:root:psu:get_fan().get_speed=0
DEBUG:root:psu:get_num_thermals()=0
DEBUG:root:psu:get_temperature=0.000
DEBUG:root:
Thermal test
DEBUG:root:Thermal-1
DEBUG:root:get_temperature=32.500
DEBUG:root:high_threshold=85.000
DEBUG:root:get name=Temp sensor 1
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:Thermal-2
DEBUG:root:get_temperature=25.500
DEBUG:root:high_threshold=84.000
DEBUG:root:get name=Temp sensor 2
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:Thermal-3
DEBUG:root:get_temperature=29.000
DEBUG:root:high_threshold=83.000
DEBUG:root:get name=Temp sensor 3
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:Thermal-4
DEBUG:root:get_temperature=27.000
DEBUG:root:high_threshold=82.000
DEBUG:root:get name=Temp sensor 4
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:
SFP test
DEBUG:root:Port-1
DEBUG:root:platform_chassis.get_serial=U31649000020
DEBUG:root:get_model()=M0OEC5402T04Z
DEBUG:root:get_name()=Ethernet0
DEBUG:root:get_rx_los=0
DEBUG:root:get_tx_fault()=0
DEBUG:root:get_tx_disable()=1
DEBUG:root:get_lp_mode=0
DEBUG:root:get_tx_disable_channel()=0x0
DEBUG:root:Port-12
DEBUG:root:platform_chassis.get_serial=U31649000020
DEBUG:root:get_model()=M0OEC5402T04Z
DEBUG:root:get_name()=Ethernet11
DEBUG:root:get_rx_los=0
DEBUG:root:get_tx_fault()=0
DEBUG:root:get_tx_disable()=1
DEBUG:root:get_lp_mode=0
DEBUG:root:get_tx_disable_channel()=0x0
**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+__all__ = ['chassis', 'eeprom', 'platform', 'psu', 'sfp', 'thermal', 'fan']
+from . import platform
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/__init__.py:2]: Add newline at EOF

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/chassis.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/chassis.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/component.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/component.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/eeprom.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/eeprom.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/component.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        Install firmware to module
+        Args:
+            image_path: A string, path to firmware image
+        Returns:
+            A boolean, True if install successfully, False if not
+        """
+        raise NotImplementedError 
+        
+
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/component.py:133]: Delete trailing blank lines and ensure file ends with newline.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/chassis.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        if not self.sfp_module_initialized:
+            self.__initialize_sfp()
+
+        try:
+            # The index will start from 1
+            sfp = self._sfp_list[index-1]
+        except IndexError:
+            sys.stderr.write("SFP index {} out of range (1-{})\n".format(
+                             index, len(self._sfp_list)))
+        return sfp
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/chassis.py:197]: Add newline at EOF

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/eeprom.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    def get_eeprom(self):
+        return self._eeprom
+
+    def get_serial(self):
+        return self._eeprom.get('0x23', "Undefined.")
+
+    def get_mac(self):
+        return self._eeprom.get('0x24', "Undefined.")
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/eeprom.py:106]: Delete trailing blank line

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/fan.py [hunk 8]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/fan.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/fan.py [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        val=self._api_helper.read_txt_file(present_path)       
+        if not self.is_psu_fan:
+            if val is not None:
+                return int(val, 10)==1
+            else:
+                return False
+            
+        else:
+            return True
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/fan.py:182]: Delete trailing blank line

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/helper.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/helper.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/helper.py [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+            p = subprocess.Popen(
+                cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            raw_data, err = p.communicate()
+            if err == '':
+                result = raw_data.strip()
+            else:
+                status = False
+        except Exception:
+            status = False
+        return status, result
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/helper.py:119]: Add newline at EOF

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/platform.py [hunk 12]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/platform.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/psu.py [hunk 13]
```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/psu.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/psu.py [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+        Returns:
+            A boolean value, True if device is operating properly, False if not
+        """
+        power_path="{}{}".format(self.cpld_path, 'psu_power_good')
+        val=self._api_helper.read_txt_file(power_path)
+        if val is not None:
+            return int(val, 10) == 1
+        else:
+            return 0
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/psu.py:231]: Remove trailing blank line

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/sfp.py [hunk 15]
```diff
@@ -0,0 +1,1229 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/sfp.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/thermal.py [hunk 16]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/thermal.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/thermal.py [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+        """
+
+        file_str = "temp{}_input".format(self.ss_index)
+        file_path = os.path.join(self.hwmon_path, file_str)
+        raw_txt = self.__read_txt_file(file_path)
+        if raw_txt is None:
+            return False
+        else:     
+            return int(raw_txt) != 0
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as5835_54x-r0/sonic_platform/thermal.py:151]: Remove trailing blank line

File: platform/broadcom/sonic-platform-modules-accton/as5835-54x/utils/accton_as5835_54x_util.py [hunk 18]
```diff
@@ -364,6 +369,73 @@ def system_ready():
         return False
     return True
 
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+PLATFORM_API2_WHL_FILE_PY2 ='sonic_platform-1.0-py2-none-any.whl'
+PLATFORM_API2_WHL_FILE_PY3 ='sonic_platform-1.0-py3-none-any.whl'
+def do_sonic_platform_install():
+    device_path = "{}{}{}{}".format(PLATFORM_ROOT_PATH, '/x86_64-accton_', PROJECT_NAME, '-r0')
+    SONIC_PLATFORM_BSP_WHL_PKG_PY2 = "/".join([device_path, PLATFORM_API2_WHL_FILE_PY2])
+    SONIC_PLATFORM_BSP_WHL_PKG_PY3 = "/".join([device_path, PLATFORM_API2_WHL_FILE_PY3])
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-accton/as5835-54x/utils/accton_as5835_54x_util.py:378]: All applications which import sonic_platform should now support Python 3 (as of a couple weeks ago). Therefore, I believe there should no longer be a need to build/install the Python 2 package. Would you mind confirming, and if so, remove all references to the Python 2 package in this PR?

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/broadcom/sonic-platform-modules-accton/as5835-54x/utils/accton_as5835_54x_util.py (hunk 18 above)
Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/as5835-54x/utils/accton_as5835_54x_util.py:378]: OK.  I will git pull new code base to check it.

**Review by jleveque — CHANGES_REQUESTED** [review]
Please fix merge conflict

**Review by jleveque — APPROVED** [review]

**Pull request merged**